tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/ImportLiferayWorkspaceOpMethods.java

```diff
@@ -85,7 +91,32 @@ public static Status execute(ImportLiferayWorkspaceOp op, ProgressMonitor pm) {
 			}
 
 			if (!initBundle && hasBundlesDir) {
-				LiferayWorkspaceUtil.addPortalRuntime(serverName);
+				Job addPortalRuntimeJob = new Job("Add Liferay Portal Runtime Job") {
+
+					@Override
+					public boolean belongsTo(Object family) {
+						return family.equals(LiferayCore.LIFERAY_JOB_FAMILY);
+					}
+
+					protected IStatus run(IProgressMonitor monitor) {
+						LiferayWorkspaceUtil.addPortalRuntime(serverName);
+
+						return org.eclipse.core.runtime.Status.OK_STATUS;
+					}
+
+				};
+
+				addPortalRuntimeJob.addJobChangeListener(
```

Comment:
There is no need for this code.  You can just call 
`addPortalRuntimeJob.join()` instead.